playbooks/installer/ospd/post.yml

```diff
@@ -19,18 +24,75 @@
 
       - block:
           - name: update file permissions
-            become: no
             file:
                 path: "{{ inventory_dir }}/id_rsa_overcloud"
                 mode: 0600
 
           - name: Update Inventory
-            become: no
             add_host:
                 name: "{{ item }}"
-                ansible_user: "heat-admin"
+                ansible_user: "{{ user }}"
             with_items: "{{ groups['overcloud_nodes'] }}"
         delegate_to: localhost
+        become: no
+
+      - shell: "cat {{ undercloud_rc }} | grep AUTH | cut -d '=' -f 2"
+        register: oc_auth
+
+      - shell: "cat {{ undercloud_rc }} | grep USERNAME | cut -d '=' -f 2"
+        register: oc_username
+
+      - shell: "cat {{ undercloud_rc }} | grep TENANT | cut -d '=' -f 2"
+        register: oc_tenant
+
+      - shell: "source {{ undercloud_rc }} && env | grep OS_PASSWORD | cut -d '=' -f 2"
+        args:
+            executable: /bin/bash
+        register: oc_password
+
+      - block:
+          - pip:
+                name: "{{ item.key }}"
+                version: "{{ item.value }}"
+                virtualenv: "{{ path_venv }}"
+            with_dict:
+                setuptools: "17.0"
+                pytz: "2016.4"
+                shade: "1.7.0"
+                pip: "8.1"
+
+          - name: update IP of overcloud nodes
+            vars:
+                ansible_python_interpreter: "/tmp/venv_server_facts/bin/python"
+            os_server_facts:
+                auth:
+                    auth_url: "{{ oc_auth.stdout }}"
+                    username: "{{ oc_username.stdout }}"
+                    password: "{{ oc_password.stdout }}"
+                    project_name: "{{ oc_tenant.stdout }}"
+
+          - debug: msg="{{ "{0}-{1}".format(item.name.split("-")[1], item.name.split("-")[2] | int ) }}"
+            with_list: "{{ openstack_servers }}"
```

Comment:
interesting... `with_list` works exactly same as `with_items` even it's not documented... but I'll fix it anyway.